packages/js/product-editor/src/hooks/use-product-edits.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { useEntityProp } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+
+type EntityEdits = {
+	[ key: string ]: unknown;
+};
+
+// Filter out the "content" and "blocks" properties of the edits since
+// we do not use these properties within the product editor and they
+// will always create false positives.
+function filterProductEdits( edits: EntityEdits ) {
+	delete edits.content;
+	delete edits.blocks;
+	return edits;
+}
+
+export function useProductEdits() {
+	const [ productId ] = useEntityProp< number >(
+		'postType',
+		'product',
+		'id'
+	);
+
+	const { edits } = useSelect(
+		( select ) => {
+			const { getEntityRecordNonTransientEdits } = select( 'core' );
+
+			const _edits = getEntityRecordNonTransientEdits(
+				'postType',
+				'product',
+				productId
+			) as EntityEdits;
+
+			return {
+				edits: filterProductEdits( _edits ),
+			};
+		},
+		[ productId ]
+	);
+
+	return {
+		hasEdits: Object.keys( edits ).length > 0,
+		edits,
```

Comment:
I think returning a few convenience functions instead of `edits` directly could make code using this easier to read...

```
hasEdit( fieldName )
getEditedValue( fieldName )
```

Reply:
Good idea!  Updated to `hasEdit` in https://github.com/woocommerce/woocommerce/pull/38430/commits/8a1d93566a95af502f6bd7e15ccdcfeed38358b0.

I've removed the `edits` for now since they're not being used elsewhere.  We can introduce them again later if they are needed and decide on any convenience methods needed at that time.